tests/integration/test_integration_all_rust.py

```diff
@@ -9,96 +9,105 @@
-)
+import websocket
 from cryptography.fernet import Fernet
-from queue import Empty, Queue
+from jose import jws
 from twisted.internet import reactor
 from twisted.internet.defer import inlineCallbacks, returnValue
 from twisted.internet.threads import deferToThread
 from twisted.trial import unittest
-from urllib.parse import urlparse
+
+from .db import (
+    DynamoDBResource,
+    base64url_encode,
+    create_message_table_ddb,
+    get_router_table,
+)
 
 app = bottle.Bottle()
 logging.basicConfig(level=logging.DEBUG)
 log = logging.getLogger(__name__)
 
 here_dir = os.path.abspath(os.path.dirname(__file__))
-root_dir = os.path.dirname(here_dir)
+tests_dir = os.path.dirname(here_dir)
+root_dir = os.path.dirname(tests_dir)
 
-DDB_JAR = os.path.join(root_dir, "tests", "ddb", "DynamoDBLocal.jar")
-DDB_LIB_DIR = os.path.join(root_dir, "tests", "ddb", "DynamoDBLocal_lib")
-DDB_PROCESS: Optional[subprocess.Popen] = None
-BT_PROCESS: Optional[subprocess.Popen] = None
-BT_DB_SETTINGS: Optional[str] = None
+DDB_JAR = os.path.join(root_dir, "tests", "integration", "ddb", "DynamoDBLocal.jar")
+DDB_LIB_DIR = os.path.join(root_dir, "tests", "integration", "ddb", "DynamoDBLocal_lib")
+DDB_PROCESS: subprocess.Popen | None = None
+BT_PROCESS: subprocess.Popen | None = None
+BT_DB_SETTINGS: str | None = None
 
 twisted.internet.base.DelayedCall.debug = True
 
 ROUTER_TABLE = os.environ.get("ROUTER_TABLE", "router_int_test")
 MESSAGE_TABLE = os.environ.get("MESSAGE_TABLE", "message_int_test")
 MSG_LIMIT = 20
 
-CRYPTO_KEY = os.environ.get("CRYPTO_KEY") or Fernet.generate_key().decode(
-    "utf-8"
-)
+CRYPTO_KEY = os.environ.get("CRYPTO_KEY") or Fernet.generate_key().decode("utf-8")
 CONNECTION_PORT = 9150
 ENDPOINT_PORT = 9160
 ROUTER_PORT = 9170
 MP_CONNECTION_PORT = 9052
 MP_ROUTER_PORT = 9072
 
 CONNECTION_BINARY = os.environ.get("CONNECTION_BINARY", "autopush_rs")
-CONNECTION_SETTINGS_PREFIX = os.environ.get(
-    "CONNECTION_SETTINGS_PREFIX", "autopush__"
-)
+CONNECTION_SETTINGS_PREFIX = os.environ.get("CONNECTION_SETTINGS_PREFIX", "autopush__")
 
-CN_SERVER: Optional[subprocess.Popen] = None
-CN_MP_SERVER: Optional[subprocess.Popen] = None
-EP_SERVER: Optional[subprocess.Popen] = None
+CN_SERVER: subprocess.Popen | None = None
+CN_MP_SERVER: subprocess.Popen | None = None
+EP_SERVER: subprocess.Popen | None = None
 MOCK_SERVER_THREAD = None
 CN_QUEUES: list = []
 EP_QUEUES: list = []
 STRICT_LOG_COUNTS = True
 
-modules = ["autoconnect", "autoconnect_common", "autoconnect_web", "autoconnect_ws", "autoconnect_ws_sm", "autoendpoint", "autopush", "autopush_common"]
+modules = [
+    "autoconnect",
+    "autoconnect_common",
+    "autoconnect_web",
+    "autoconnect_ws",
+    "autoconnect_ws_sm",
+    "autoendpoint",
+    "autopush",
+    "autopush_common",
+]
 log_string = [f"{x}=trace" for x in modules]
-RUST_LOG = ",".join(log_string)+",error"
+RUST_LOG = ",".join(log_string) + ",error"
 
 
-def get_free_port() -> int:
+def get_free_port() -> Any:
```

Comment:
Are there instances where the `port` is not an integer?

Reply:
sigh*
<img width="783" alt="image" src="https://github.com/mozilla-services/autopush-rs/assets/3422688/c2d79f59-9f04-4886-b465-304dbe173836">
mypy gymnastics see https://github.com/mozilla-services/autopush-rs/pull/518/commits/12ae9f015398b0ba897e8a7aedf52d223e62ea57